lib/iris/experimental/regrid.py

```diff
@@ -114,6 +115,29 @@ def _get_xy_coords(cube):
     return x_coord, y_coord
 
 
+def regrid_bilinear_rectilinear_src_and_grid(src, grid,
+                                             extrapolation_mode='mask'):
+    """
+    Return a new Cube that is the result of regridding the source Cube
+    onto the grid of the grid Cube using bilinear interpolation.
+
+    .. note::
+
+        This routine has now been removed, and should be replaced by a call to
+        :meth:`iris.cube.Cube.regrid`, passing an instance of
+        :class:`iris.analysis.Linear` as the 'scheme' argument.
+
+    For example, replace
+        "regrid_bilinear_rectilinear_src_and_grid(src, grid, mode)", with
+        "src.regrid(grid, iris.analysis.Linear(mode))".
+
+    """
+    msg = ("This routine has now been removed, and should be replaced by "
+           "a call to iris.cube.Cube.regrid, passing an instance of "
+           "iris.analysis.Linear as the 'scheme' argument.")
+    raise iris.exceptions.IrisError(msg)
+
+
```

Comment:
@pp-mo I understand you're intent here, but at the end of the day this is experimental functionality, so in my opinion there's no need to raise this kind of exception. The addition to the `whatsnew` provides a sufficient breadcrumb trail for the interested user.

We definitely don't want to start accruing such exceptions within `experimental` for code that's been removed, re-branded, or retired. That's the deal; it's experimental, stuff can change!